x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/TokensInvalidationResult.java

```diff
@@ -0,0 +1,139 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.security.authc.support;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * The result of attempting to invalidate one or multiple tokens. The result contains information about:
+ * <ul>
+ * <li>how many of the tokens were actually invalidated</li>
+ * <li>how many tokens are not invalidated in this request because they were already invalidated</li>
+ * <li>how many tokens were not invalidated because of an error and what the error was</li>
+ * </ul>
+ */
+public class TokensInvalidationResult implements ToXContentObject {
+
+    private final List<String> invalidatedTokens;
+    private final List<String> previouslyInvalidatedTokens;
+    private final List<ElasticsearchException> errors;
+    private final int attemptCount;
+
+    public TokensInvalidationResult(List<String> invalidatedTokens, List<String> previouslyInvalidatedTokens,
+                                    @Nullable List<ElasticsearchException> errors, int attemptCount) {
+        Objects.requireNonNull(invalidatedTokens, "invalidated_tokens must be provided");
+        this.invalidatedTokens = invalidatedTokens;
+        Objects.requireNonNull(previouslyInvalidatedTokens, "previously_invalidated_tokens must be provided");
+        this.previouslyInvalidatedTokens = previouslyInvalidatedTokens;
+        if (null != errors) {
+            this.errors = errors;
+        } else {
+            this.errors = Collections.emptyList();
+        }
+        this.attemptCount = attemptCount;
+    }
+
+    public static TokensInvalidationResult emptyResult() {
+        return new TokensInvalidationResult(Collections.emptyList(), Collections.emptyList(), Collections.emptyList(), 0);
+    }
+
+    public static TokensInvalidationResult emptyResultWithCounter(int attemptCount) {
+        return new TokensInvalidationResult(Collections.emptyList(), Collections.emptyList(), Collections.emptyList(), attemptCount);
+    }
+
+    public List<String> getInvalidatedTokens() {
+        return invalidatedTokens;
+    }
+
+    public List<String> getPreviouslyInvalidatedTokens() {
+        return previouslyInvalidatedTokens;
+    }
+
+    public List<ElasticsearchException> getErrors() {
+        return errors;
+    }
+
+    public int getAttemptCount() {
+        return attemptCount;
+    }
+
+    public static void writeTo(TokensInvalidationResult result, StreamOutput out) throws IOException {
```

Comment:
instead of the static methods, I think we prefer using Writeable now. So you would have a non static writeTo method and then a constructor that accepts a streaminput